java/query/src/main/java/sleeper/query/model/LeafPartitionQuery.java

```diff
@@ -69,122 +105,121 @@ public List<String> getFiles() {
+                .leafPartitionId(leafPartitionId)
+                .partitionRegion(partitionRegion)
+                .files(files);
     }
 
     @Override
-    public boolean equals(Object obj) {
-        if (this == obj) {
+    public boolean equals(Object object) {
+        if (this == object) {
             return true;
         }
-        if (obj == null) {
-            return false;
-        }
-        if (getClass() != obj.getClass()) {
+        if (object == null || getClass() != object.getClass()) {
             return false;
         }
-        LeafPartitionQuery other = (LeafPartitionQuery) obj;
-        if (!super.equals(other)) {
-            return false;
-        }
-        if (!Objects.equals(this.subQueryId, other.subQueryId)) {
-            return false;
-        }
-        if (!Objects.equals(this.leafPartitionId, other.leafPartitionId)) {
-            return false;
-        }
-        if (!Objects.equals(this.partitionRegion, other.partitionRegion)) {
-            return false;
+        LeafPartitionQuery that = (LeafPartitionQuery) object;
+        return Objects.equals(tableId, that.tableId)
+                && Objects.equals(queryId, that.queryId)
+                && Objects.equals(subQueryId, that.subQueryId)
+                && Objects.equals(regions, that.regions)
+                && Objects.equals(processingConfig, that.processingConfig)
+                && Objects.equals(leafPartitionId, that.leafPartitionId)
+                && Objects.equals(partitionRegion, that.partitionRegion)
+                && Objects.equals(files, that.files);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(tableId, queryId, subQueryId, regions, processingConfig, leafPartitionId, partitionRegion, files);
+    }
+
+    @Override
+    public String toString() {
+        return "LeafPartitionQuery{" +
+                "tableId='" + tableId + '\'' +
+                ", queryId='" + queryId + '\'' +
+                ", subQueryId='" + subQueryId + '\'' +
+                ", regions=" + regions +
+                ", processingConfig=" + processingConfig +
+                ", leafPartitionId='" + leafPartitionId + '\'' +
+                ", partitionRegion=" + partitionRegion +
+                ", files=" + files +
+                '}';
+    }
+
+    public static final class Builder {
+        private String tableId;
+        private String queryId;
+        private String subQueryId;
+        private List<Region> regions;
+        private QueryProcessingConfig processingConfig;
+        private String leafPartitionId;
+        private Region partitionRegion;
+        private List<String> files;
+
+        private Builder() {
         }
-        if (!Objects.equals(new HashSet<>(this.files), new HashSet<>(other.files))) {
-            return false;
+
+        public Builder parentQuery(Query parentQuery) {
+            return queryId(parentQuery.getQueryId())
+                    .regions(parentQuery.getRegions())
```

Comment:
I'm not sure we want to take the regions from the parent query. That means that the user has to remember to call regions after parentQuery if they want to supply their own regions, and not use the ones from the parent query (which is the usual case). If they call them the other way around the regions from the parent query will overwrite the regions they've already provided.

Reply:
Good point, I'll change that.